Algorithms/Encoders/SoundexEncoder.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Linq;
+
+namespace Algorithms.Encoders
+{
+    /// <summary>
+    /// Class for Soundex encoding strings.
+    /// </summary>
+    public class SoundexEncoder
+    {
+        /// <summary>
+        /// Creates a new instance of the SoundexEncoder.
+        /// </summary>
+        public SoundexEncoder() {}
+
+        /// <summary>
+        /// Encodes a string using the Soundex Algorithm
+        /// </summary>
+        /// <param name="text">The string to encode</param>
+        /// <returns>The Soundex encoded string (one uppercase character and three digits)</returns>
+        public string Encode(string text)
+        {
+            string soundex = $"{text.ToUpper()[0]}";
```

Comment:
Codacy found an issue: [Define the locale to be used in this string operation.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6311443)